ReactAndroid/src/main/java/com/facebook/react/ReactNativeHost.java

```diff
@@ -34,13 +35,23 @@ protected ReactNativeHost(Application application) {
   /** Get the current {@link ReactInstanceManager} instance, or create one. */
   public ReactInstanceManager getReactInstanceManager() {
     if (mReactInstanceManager == null) {
+      // Ensure {@link ReactPerformanceLogger} exists before we create {@link ReactInstanceManager}.
+      getReactPerformanceLogger();
       ReactMarker.logMarker(ReactMarkerConstants.GET_REACT_INSTANCE_MANAGER_START);
       mReactInstanceManager = createReactInstanceManager();
       ReactMarker.logMarker(ReactMarkerConstants.GET_REACT_INSTANCE_MANAGER_END);
     }
     return mReactInstanceManager;
   }
 
+  /** Get the current {@link ReactPerformanceLogger} instance, or create one. */
+  public ReactPerformanceLogger getReactPerformanceLogger() {
+    if (mReactPerformanceLogger == null) {
+      mReactPerformanceLogger = new ReactPerformanceLogger();
```

Comment:
Why can't this be a singleton ? I am not sure if we are using this instance variable anywhere else.